src/test/java/seedu/recipe/logic/commands/RecipeDescriptorTest.java

```diff
@@ -1,36 +1,202 @@
 
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import seedu.recipe.logic.util.RecipeDescriptor;
+import seedu.recipe.model.recipe.Name;
+import seedu.recipe.model.recipe.Recipe;
+import seedu.recipe.model.recipe.RecipeDuration;
+import seedu.recipe.model.recipe.RecipePortion;
+import seedu.recipe.model.recipe.Step;
+import seedu.recipe.model.recipe.ingredient.Ingredient;
+import seedu.recipe.model.recipe.ingredient.IngredientBuilder;
+import seedu.recipe.model.recipe.ingredient.IngredientInformation;
+import seedu.recipe.model.recipe.ingredient.IngredientQuantity;
+import seedu.recipe.model.tag.Tag;
+
 public class RecipeDescriptorTest {
+    private static final Name TEST_NAME = new Name("Cacio e Pepe");
+    private static final RecipeDuration TEST_DURATION = RecipeDuration.of("15 minutes");
+    private static final RecipePortion TEST_PORTION = RecipePortion.of("1 - 2 servings");
+    private static final Set<Tag> TEST_TAGS = Set.of(new Tag("Italian"), new Tag("Carbs"));
+    private static final HashMap<Ingredient, IngredientInformation> TEST_INGREDIENT_TABLE;
+    private static final IngredientBuilder TEST_BUILDER;
+    private static final List<IngredientBuilder> TEST_INGREDIENT_LIST;
+    private static final List<Step> TEST_STEPS;
+    static {
+        TEST_INGREDIENT_TABLE = new HashMap<>();
+        TEST_INGREDIENT_TABLE.put(
+                Ingredient.of("Pasta"),
+                new IngredientInformation(
+                        IngredientQuantity.of("100 g"), null, new String[]{}, new Ingredient[]{}
+                )
+        );
+
+        TEST_BUILDER = new IngredientBuilder("-n Cheese -a 1 wedge");
+        TEST_INGREDIENT_LIST = List.of(TEST_BUILDER);
+
+        TEST_STEPS = List.of(new Step("Grate the cheese."));
+    }
+    private RecipeDescriptor testDescriptor;
+
+    @BeforeEach
+    public void setTestDescriptor() {
+        testDescriptor = new RecipeDescriptor();
+    }
+
+    @Test
+    public void defaultConstructor_initOkay() {
+        assertDoesNotThrow(() -> new RecipeDescriptor());
+    }
+
+    @Test
+    public void defaultConstructor_anyFieldEdited_false() {
+        assertFalse(testDescriptor.isAnyFieldEdited());
+    }
+
+    @Test
+    public void defaultConstructor_toRecipe_nameBlank() {
+        Recipe expected = new Recipe(new Name("BLANK RECIPE"));
+        assertEquals(testDescriptor.toRecipe(), expected);
+    }
+
+    @Test
+    public void defaultConstructor_get_fieldsEmpty() {
```

Comment:
Not clear enough, `defaultConstructor_getFields_allFieldsEmpty`

Reply:
fixed